homeassistant/components/tts/__init__.py

```diff
@@ -245,12 +248,14 @@ def async_register_engine(self, engine, provider, config):
         self.providers[engine] = provider
 
     @asyncio.coroutine
-    def async_get_url(self, engine, message, cache=None):
+    def async_get_url(self, engine, message, cache=None, language=None):
         """Get URL for play message.
 
         This method is a coroutine.
         """
-        msg_hash = hashlib.sha1(bytes(message, 'utf-8')).hexdigest()
+        msg_hash = hashlib.sha1(bytes(
```

Comment:
We have no db to store other info. In future I will make the clear_cache selective for provider and language. It is also more userfriendly to have a bit more info about the file. So stuff like ssml should be hash to but provider/language are good selector for do additional stuff.